src/simd.jl

```diff
@@ -0,0 +1,576 @@
+    F = SVec{36, T}((T(1), T(2), T(2), T(1), T(2), T(1), T(2), T(4), T(4), T(2), T(4), T(2),
+                     T(2), T(4), T(4), T(2), T(4), T(2), T(1), T(2), T(2), T(1), T(2), T(1),
+                     T(2), T(4), T(4), T(2), T(4), T(2), T(1), T(2), T(2), T(1), T(2), T(1)))
+    D = SVec{36, T}(get_data(S))
+    DD = D * D; FDD = F * DD; sFDD = sum(FDD)
+    return sqrt(sFDD)
+end
+
+# dcontract
+@inline dcontract{dim, T <: SIMDTypes}(S1::SecondOrderTensor{dim, T}, S2::SecondOrderTensor{dim, T}) = dcontract(promote(S1, S2)...)
+
+@inline function dcontract{T <: SIMDTypes}(S1::Tensor{2, 3, T}, S2::Tensor{2, 3, T})
+    D1 = SVec{9, T}(get_data(S1))
+    D2 = SVec{9, T}(get_data(S2))
+    D1D2 = D1 * D2
+    return sum(D1D2)
+end
+
+@inline function dcontract{T <: SIMDTypes}(S1::SymmetricTensor{2, 3, T}, S2::SymmetricTensor{2, 3, T})
+    D1 = SVec{6, T}(get_data(S1))
+    D2 = SVec{6, T}(get_data(S2))
+    F  = SVec{6, T}((T(1), T(2), T(2), T(1), T(2), T(1)))
+    D1D2 = D1 * D2; FD1D2 = F * D1D2
+    return sum(FD1D2)
+end
+
+@inline function dcontract{T <: SIMDTypes}(S1::Tensor{4, 1, T}, S2::Tensor{2, 1, T})
+    @inbounds begin
+        D1 = get_data(S1)
+        D2 = get_data(S2)
+        D11 = SVec{4, T}((D1[1], ))
+        r  = D11 * D2[1]
+        return Tensor{2, 1}((r[1], ))
+    end
+end
+
+@inline function dcontract{T <: SIMDTypes}(S1::Tensor{4, 2, T}, S2::Tensor{2, 2, T})
+    @inbounds begin
+        D1 = get_data(S1)
+        D2 = get_data(S2)
+
+        D11 = SVec{4, T}((D1[1],  D1[2],  D1[3],  D1[4]))
+        D12 = SVec{4, T}((D1[5],  D1[6],  D1[7],  D1[8]))
+        D13 = SVec{4, T}((D1[9],  D1[10], D1[11], D1[12]))
+        D14 = SVec{4, T}((D1[13], D1[14], D1[15], D1[16]))
+
+        r  = D11 * D2[1]
+        r += D12 * D2[2]
+        r += D13 * D2[3]
+        r += D14 * D2[4]
+
+        return Tensor{2, 2}((r[1], r[2], r[3], r[4]))
+    end
+end
+
+@inline function dcontract{T <: SIMDTypes}(S1::Tensor{4,3,T}, S2::Tensor{2,3,T})
+    @inbounds begin
+        D1 = get_data(S1)
+        D2 = get_data(S2)
+
+        D11 = SVec{9, T}((D1[1],  D1[2],  D1[3],  D1[4],  D1[5],  D1[6],  D1[7],  D1[8],  D1[9]))
+        D12 = SVec{9, T}((D1[10], D1[11], D1[12], D1[13], D1[14], D1[15], D1[16], D1[17], D1[18]))
+        D13 = SVec{9, T}((D1[19], D1[20], D1[21], D1[22], D1[23], D1[24], D1[25], D1[26], D1[27]))
+        D14 = SVec{9, T}((D1[28], D1[29], D1[30], D1[31], D1[32], D1[33], D1[34], D1[35], D1[36]))
+        D15 = SVec{9, T}((D1[37], D1[38], D1[39], D1[40], D1[41], D1[42], D1[43], D1[44], D1[45]))
+        D16 = SVec{9, T}((D1[46], D1[47], D1[48], D1[49], D1[50], D1[51], D1[52], D1[53], D1[54]))
+        D17 = SVec{9, T}((D1[55], D1[56], D1[57], D1[58], D1[59], D1[60], D1[61], D1[62], D1[63]))
+        D18 = SVec{9, T}((D1[64], D1[65], D1[66], D1[67], D1[68], D1[69], D1[70], D1[71], D1[72]))
+        D19 = SVec{9, T}((D1[73], D1[74], D1[75], D1[76], D1[77], D1[78], D1[79], D1[80], D1[81]))
+
+        r  = D11 * D2[1]
```

Comment:
I guess otherwise it needs to accumulate it to the same register and wait for the previous operation to be done. This was it can do stuff in "parallell".